src/View.elm

```diff
@@ -63,10 +63,90 @@ timelineView statuses label =
         ]
 
 
+draftView : Model -> Html Msg
+draftView model =
+    let
+        hasSpoiler =
+            case model.draft.spoiler_text of
+                Nothing ->
+                    False
+
+                Just _ ->
+                    True
+    in
+        div [ class "col-md-3" ]
+            [ div [ class "panel panel-default" ]
+                [ div [ class "panel-heading" ] [ text "Post a message" ]
+                , div [ class "panel-body" ]
+                    [ Html.form [ class "form", onSubmit SubmitDraft ]
+                        [ div [ class "form-group checkbox" ]
+                            [ label []
+                                [ input
+                                    [ type_ "checkbox"
+                                    , onCheck <| DraftEvent << ToggleSpoiler
+                                    , checked hasSpoiler
+                                    ]
+                                    []
+                                , text " Add a spoiler"
+                                ]
+                            ]
+                        , if hasSpoiler then
+                            div [ class "form-group" ]
+                                [ label [ for "spoiler" ] [ text "Spoiler" ]
+                                , textarea
+                                    [ id "spoiler"
+                                    , class "form-control"
+                                    , rows 5
+                                    , placeholder "This text will always be visible."
```

Comment:
Oh boy, we need something clearer here. It took me some time to get it even after looking at your gif.
Maybe just renaming "Spoiler" to "Visible part" and "Status" to "Hidden part" or "Spoiler part", something along the lines?